internal/hydra/device.go

```diff
@@ -0,0 +1,231 @@
+	return v.value
+}
+
+func (v *NullableAcceptDeviceUserCodeRequest) Set(val *AcceptDeviceUserCodeRequest) {
+	v.value = val
+	v.isSet = true
+}
+
+func (v NullableAcceptDeviceUserCodeRequest) IsSet() bool {
+	return v.isSet
+}
+
+func (v *NullableAcceptDeviceUserCodeRequest) Unset() {
+	v.value = nil
+	v.isSet = false
+}
+
+func NewNullableAcceptDeviceUserCodeRequest(val *AcceptDeviceUserCodeRequest) *NullableAcceptDeviceUserCodeRequest {
+	return &NullableAcceptDeviceUserCodeRequest{value: val, isSet: true}
+}
+
+func (v NullableAcceptDeviceUserCodeRequest) MarshalJSON() ([]byte, error) {
+	return json.Marshal(v.value)
+}
+
+func (v *NullableAcceptDeviceUserCodeRequest) UnmarshalJSON(src []byte) error {
+	v.isSet = true
+	return json.Unmarshal(src, &v.value)
+}
+
+type ApiAcceptUserCodeRequestRequest struct {
+	ctx                         context.Context
+	ApiService                  OAuth2Api
+	deviceChallenge             *string
+	acceptDeviceUserCodeRequest *AcceptDeviceUserCodeRequest
+}
+
+func (r ApiAcceptUserCodeRequestRequest) DeviceChallenge(deviceChallenge string) ApiAcceptUserCodeRequestRequest {
+	r.deviceChallenge = &deviceChallenge
+	return r
+}
+
+func (r ApiAcceptUserCodeRequestRequest) AcceptDeviceUserCodeRequest(acceptDeviceUserCodeRequest AcceptDeviceUserCodeRequest) ApiAcceptUserCodeRequestRequest {
+	r.acceptDeviceUserCodeRequest = &acceptDeviceUserCodeRequest
+	return r
+}
+
+func (r ApiAcceptUserCodeRequestRequest) Execute() (*hClient.OAuth2RedirectTo, *http.Response, error) {
+	return r.ApiService.AcceptUserCodeRequestExecute(r)
+}
+
+/*
+AcceptUserCodeRequest Accepts a device grant user_code request
+
+Accepts a device grant user_code request
+
+	@param ctx context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
+	@return ApiAcceptUserCodeRequestRequest
+*/
+func (a *DeviceApiService) AcceptUserCodeRequest(ctx context.Context) ApiAcceptUserCodeRequestRequest {
+	return ApiAcceptUserCodeRequestRequest{
+		ApiService: a,
+		ctx:        ctx,
+	}
+}
+
+// Execute executes the request
+//
+//	@return OAuth2RedirectTo
+func (a *DeviceApiService) AcceptUserCodeRequestExecute(r ApiAcceptUserCodeRequestRequest) (*hClient.OAuth2RedirectTo, *http.Response, error) {
+	body, err := json.Marshal(r.acceptDeviceUserCodeRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	localBasePath, err := a.client.GetConfig().ServerURLWithContext(r.ctx, "OAuth2ApiService.AcceptUserCodeRequest")
+	if err != nil {
+		return nil, nil, err
+	}
+	url := localBasePath + "/admin/oauth2/auth/requests/device/accept?challenge=" + *r.deviceChallenge
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, nil, reportError("error when constructing request: %s", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
```

Comment:
is it worth using a shared client? ideally we could even wrap it with the `otel` http client to spread traces (which can help with troubleshooting)

Reply:
Good point, I changed it